core/src/main/java/org/infinispan/container/offheap/StripedLock.java

```diff
@@ -38,17 +37,17 @@ public ReadWriteLock getLockWithOffset(int offset) {
     * Locks all write locks.  Ensure that {@link StripedLock#unlockAll()} is called in a proper finally block
     */
    public void lockAll() {
-      for (ReadWriteLock rwLock : locks) {
-         rwLock.writeLock().lock();
+      for (StampedLock rwLock : locks) {
+         rwLock.asWriteLock().lock();
       }
    }
 
    /**
     * Unlocks all write locks, useful after {@link StripedLock#lockAll()} was invoked.
     */
    void unlockAll() {
-      for (ReadWriteLock rwLock : locks) {
-         rwLock.writeLock().unlock();
+      for (StampedLock rwLock : locks) {
+         rwLock.asWriteLock().unlock();
```

Comment:
Too bad `unstampedUnlockWrite()` is package-private :)

Reply:
We would be able to avoid the Lock allocation :) The instance isn't that large though.